### PR TITLE
chore: add more filter relevant UT in analysis_test.go

### PR DESCRIPTION
**Pull request by panpan0000**
Closes #NA

It's the UT code from closed PR https://github.com/k8sgpt-ai/k8sgpt/pull/402
I thinks the UT is worthy, so contribute them again.

## 📑 Description

3 new cases to test:

- The user adds --filter in his analysis command
- We get the actives_filters (stored in the configuration file)
- Neither of them is specified and we use the core analyzers

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: pkg/analysis/analysis_test.go [hunk 1]
```diff
@@ -14,14 +14,127 @@ limitations under the License.
...
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"strings"
+	"testing"
 )
 
+// sub-function
+func analysis_RunAnalysis_Filter_Tester(t *testing.T, filterFlag string) []common.Result {
```
Comment (AlexsJones) [inline pkg/analysis/analysis_test.go:34]: Can we please make the function signatures snake_case lower cased?

**Review by AlexsJones — COMMENTED** [review]
Looks great, thank you - just a comment around signature casing

**Review by AlexsJones — COMMENTED** [review]
Looks great, thank you - just a comment around signature casing

File: pkg/analysis/analysis_test.go (hunk 1 above)
Comment (panpan0000) [reply pkg/analysis/analysis_test.go]: fixed, thanks

**Comment by panpan0000** [comment]
done, thank Alex

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**